Core/Generators/Rust/RustGenerator.cs

```diff
@@ -834,6 +858,29 @@ private bool NeedsLifetime(Definition d, OwnershipType ot = OwnershipType.Borrow
             return _needsLifetime[d.Name];
         }
 
+        private bool IsFixedSizeStruct(StructDefinition d)
+        {
+            if (_isFixedSizeStruct.ContainsKey(d.Name))
+            {
+                return _isFixedSizeStruct[d.Name];
+            }
+
+            var isPrimitive = d.Fields.All((f) =>
+                f.Type switch
+                {
+                    DefinedType dt => Schema.Definitions[dt.Name] switch
+                    {
+                        StructDefinition sd => IsFixedSizeStruct(sd),
+                        EnumDefinition ed => true,
+                        _ => false
+                    },
+                    ScalarType st => st.IsFixedScalar(),
+                    _ => false
+                });
+
+            return _isFixedSizeStruct[d.Name] = isPrimitive;
+        }
+
```

Comment:
I would make this an extension method for the `StructDefinition` type since its reusable across generators.